pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -35,8 +35,10 @@ var (
 	adsDebug = os.Getenv("PILOT_DEBUG_ADS") != "0"
 
 	// adsClients reflect active gRPC channels, for both ADS and EDS.
-	adsClients      = map[string]*XdsConnection{}
-	adsClientsMutex sync.RWMutex
+	adsClients = map[string]*XdsConnection{}
+	// Has map of sidecar ID to the same connection struct as above
```

Comment:
Note there is a strange corner case of envoy restart where the 'old' envoy connects after the 'new'/restarted envoy. 
Even have a regression test for this - it happened.

So we can't rely on time of last connection. Maybe (in a different PR) we can make it a []*XdsConnection - it'll also tell
us how many connections each sidecar has, good for metrics.